File: elixir/bob/bob.exs
defmodule Bob do
  def hey(input) do
    cond do
        String.ends_with?(input, "?") -> "Sure."
        String.match?(input, ~r/^\s*$/) -> "Fine. Be that way!"
        String.upcase(input) == input and String.downcase(input) != input -> "Whoa, chill out!"
        true ->  "Whatever."
    end
  end
end

File: scala/bob/src/main/scala/bob.scala
class Bob {
  def hey(string:String):String = {

    val upper = string.toUpperCase
    val lower = string.toLowerCase
    val question = """.*\?""".r
    val silence = """^\s*$""".r
    
    string match {
      case string if string.equals(upper) && !string.equals(lower) => "Whoa, chill out!"
      case question(_*) => "Sure."
      case silence(_*) => "Fine. Be that way!"
      case _ => "Whatever."
    }
  }
}
